src/services/pact-verifier.service.ts

```diff
@@ -7,27 +6,29 @@ import { PactProviderOptions } from '../interfaces/pact-provider-module-options.
 @Injectable()
 export class PactVerifierService {
   public constructor(
-    @Inject(PactModuleProviders.ProviderOptions) private readonly options: PactProviderOptions,
-    @Inject(PactModuleProviders.PactVerifier) private readonly verifier: Verifier
+    @Inject(PactModuleProviders.PactVerifier) private readonly verifier: Verifier,
+    @Inject(PactModuleProviders.ProviderOptions) private readonly options: PactProviderOptions
   ) {}
 
   public async verify(app: INestApplication): Promise<any> {
-    const host = this.options.providerHost || 'localhost';
-    await app.listen(await getPort(), host);
+    const providerUrl = new URL(this.options.providerBaseUrl);
+    let results: string;
 
-    const appUrl = await app.getUrl();
+    await app.listen(providerUrl.port || 80, providerUrl.hostname);
 
-    const results = await this.verifier.verifyProvider({
-      ...this.options,
-      providerBaseUrl: appUrl,
-    });
+    // we need the try-catch here because the verifier will throw an error
+    // and the application will not close if the error is not caught
+    try {
+      results = await this.verifier.verifyProvider();
+    } catch (e) {
+      // do nothing
```

Comment:
Great catch!!

However, this isn't the right fix. This fix will mean that a failing verification won't fail the test suite 

Better to do:

```
const results = await this.verifier.verifyProvider().finally(() => app.close());
return results;
```

or more simply:

```
return this.verifier.verifyProvider().finally(() => app.close());
```

As an aside, in general, it's never a good idea to have empty catch blocks - as it can make root causing issues very difficult.

Reply:
very good point. Updated it to your simplified version, but left the small comment.